diff --git a/sites/all/modules/contrib/biblio/modules/crossref/biblio.crossref.client.php b/sites/all/modules/contrib/biblio/modules/crossref/biblio.crossref.client.php
index 6c4d1ac..fb4272b 100644
--- a/sites/all/modules/contrib/biblio/modules/crossref/biblio.crossref.client.php
+++ b/sites/all/modules/contrib/biblio/modules/crossref/biblio.crossref.client.php
@@ -291,8 +291,13 @@ class BiblioCrossRefClient
           }
           break;
         case 'resource':
-          if ($this->doi_data) {
-              $this->_set_data('biblio_url', $data);
+          static $finished = FALSE;
+          if ($this->doi_data && !$finished) {
+              if(strlen($this->node['biblio_url']) && (strpos($data, 'http://')!==FALSE || strpos($data, 'https://'))){
+                $finished = TRUE;
+              } else {
+                $this->_set_data('biblio_url', $data);
+              }
           }
           break;
