diff --git a/sites/all/modules/contrib/biblio/biblio.module b/sites/all/modules/contrib/biblio/biblio.module
index 0660411..e350b9a 100644
--- a/sites/all/modules/contrib/biblio/biblio.module
+++ b/sites/all/modules/contrib/biblio/biblio.module
@@ -1894,7 +1894,7 @@ function _biblio_citekey_print($citekey) {
     $style = biblio_get_style();
     $base = variable_get('biblio_base', 'biblio');
     $node = node_load($nid->nid);
-    return theme('biblio_style', array('node' => $node, 'base' => $base, 'style' => $style));
+    return theme('biblio_style', array('node' => $node, 'base' => $base, 'style_name' => $style));
   }
   else {
     return t("Citekey @cite not found", array('@cite' => $citekey));
