diff --git a/plugins/FeedsTermProcessor.inc b/plugins/FeedsTermProcessor.inc
index b336096..0e74544 100644
--- a/plugins/FeedsTermProcessor.inc
+++ b/plugins/FeedsTermProcessor.inc
@@ -97,16 +97,12 @@ class FeedsTermProcessor extends FeedsProcessor {
               $parent_tid = $term->tid;
             }
           }
-          if (!empty($parent_tid)) {
-            $target_term->parent[] = $parent_tid;
-          }
-          else {
-            $target_term->parent[] = 0;
+          if ($parent_tid) {
+            if(!isset($target_term->parent) || array_search($parent_tid, $target_term->parent) === FALSE){
+              $target_term->parent[] = $parent_tid;
+            }
           }
         }
-        else {
-          $target_term->parent[] = 0;
-        }
         break;
       case 'parentguid':
         // value is parent_guid field value
@@ -114,8 +110,11 @@ class FeedsTermProcessor extends FeedsProcessor {
           ->fields('feeds_item', array('entity_id'))
           ->condition('entity_type', $this->entityType());
         $parent_tid = $query->condition('guid', $value)->execute()->fetchField();
-        $target_term->parent[] = ($parent_tid) ? $parent_tid : 0;
-
+        if($parent_tid){
+          if(!isset($target_term->parent) || array_search($parent_tid, $target_term->parent) === FALSE){
+            $target_term->parent[] = $parent_tid;
+          }
+        }
         break;
       case 'weight':
         if (!empty($value)) {
