diff --git a/sites/all/modules/contrib/field_collection/field_collection.module b/sites/all/modules/contrib/field_collection/field_collection.module
index f27b2d5..fe588e4 100644
--- a/sites/all/modules/contrib/field_collection/field_collection.module
+++ b/sites/all/modules/contrib/field_collection/field_collection.module
@@ -1912,42 +1912,44 @@ function field_collection_feeds_set_target($source, $entity, $target, $value) {
   $language = (isset($entity->language)) ? $entity->language : LANGUAGE_NONE;
 
   foreach ($value as $v) {
-    if (isset($field[$language][$delta]['entity'])) {
-      $field_collection_item = $field[$language][$delta]['entity'];
-    }
-    elseif (isset($field[$language][$delta]['value'])) {
-      $field_collection_item = field_collection_item_load($field[$language][$delta]['value']);
-    }
-    if (empty($field_collection_item)) {
-      $field_collection_item = entity_create('field_collection_item', array('field_name' => $target));
-    }
-
-    // Hack: feeds mapping callbacks (at least text_feeds_set_target), seem to
-    // want an empty array for a Field Collection item target, to be sure the
-    // item is iterated over at least once.
-    unset($field_collection_item->$sub_target);
-
-    // let's see if the field in the field collection has multiple cardinality
-    // in case the field is complex type such as field_url:title, field_url:link
-    $sub_field = explode(':', $sub_target);
-    $sub_field_info = field_info_field($sub_field[0]);
-    if (isset($_sub_targets[$sub_target]['callback']) && function_exists($_sub_targets[$sub_target]['callback'])) {
-      $callback = $_sub_targets[$sub_target]['callback'];
-      if($sub_field_info['cardinality'] == 1) {
-        $callback($source, $field_collection_item, $sub_target, $v);
+    if(strlen($v)){
+      if (isset($field[$language][$delta]['entity'])) {
+        $field_collection_item = $field[$language][$delta]['entity'];
       }
-      else {
-        $callback($source, $field_collection_item, $sub_target, $value);
+      elseif (isset($field[$language][$delta]['value'])) {
+        $field_collection_item = field_collection_item_load($field[$language][$delta]['value']);
+      }
+      if (empty($field_collection_item)) {
+        $field_collection_item = entity_create('field_collection_item', array('field_name' => $target));
+      }
+  
+      // Hack: feeds mapping callbacks (at least text_feeds_set_target), seem to
+      // want an empty array for a Field Collection item target, to be sure the
+      // item is iterated over at least once.
+      unset($field_collection_item->$sub_target);
+  
+      // let's see if the field in the field collection has multiple cardinality
+      // in case the field is complex type such as field_url:title, field_url:link
+      $sub_field = explode(':', $sub_target);
+      $sub_field_info = field_info_field($sub_field[0]);
+      if (isset($_sub_targets[$sub_target]['callback']) && function_exists($_sub_targets[$sub_target]['callback'])) {
+        $callback = $_sub_targets[$sub_target]['callback'];
+        if($sub_field_info['cardinality'] == 1) {
+          $callback($source, $field_collection_item, $sub_target, $v);
+        }
+        else {
+          $callback($source, $field_collection_item, $sub_target, $value);
+        }
+      }
+      $field[$language][$delta]['entity'] = $field_collection_item;
+      $field[$language][$delta]['value'] = $field_collection_item->item_id;
+      unset($field_collection_item);
+      if ($info['cardinality'] == 1) {
+        break;
+      }
+      if ($sub_field_info['cardinality'] == 1) {
+        $delta++;
       }
-    }
-    $field[$language][$delta]['entity'] = $field_collection_item;
-    $field[$language][$delta]['value'] = $field_collection_item->item_id;
-    unset($field_collection_item);
-    if ($info['cardinality'] == 1) {
-      break;
-    }
-    if ($sub_field_info['cardinality'] == 1) {
-      $delta++;
     }
   }
   $entity->{$target} = $field;
