diff --git a/views/media_default.view.inc b/views/media_default.view.inc
index f4ba4b9..f440b64 100644
--- a/views/media_default.view.inc
+++ b/views/media_default.view.inc
@@ -28,7 +28,7 @@ $handler->display->display_options['query']['options']['query_tags'] = array(
   0 => 'media_browser',
 );
 $handler->display->display_options['exposed_form']['type'] = 'basic';
-$handler->display->display_options['exposed_form']['options']['autosubmit'] = TRUE;
+$handler->display->display_options['exposed_form']['options']['reset_button'] = FALSE;
 $handler->display->display_options['pager']['type'] = 'full';
 $handler->display->display_options['pager']['options']['items_per_page'] = '25';
 $handler->display->display_options['pager']['options']['offset'] = '0';
