diff --git a/sites/all/modules/contrib/biblio/modules/pubmed/EntrezClient.php b/sites/all/modules/contrib/biblio/modules/pubmed/EntrezClient.php
index 9c88a9244..0f4dd6f36 100644
--- a/sites/all/modules/contrib/biblio/modules/pubmed/EntrezClient.php
+++ b/sites/all/modules/contrib/biblio/modules/pubmed/EntrezClient.php
@@ -9,7 +9,7 @@ class BiblioEntrezClient
 {
   const DEFAULT_DATABASE = 'pubmed';
 
-  const BASE_URL = 'http://eutils.ncbi.nlm.nih.gov/entrez/eutils/';
+  const BASE_URL = 'https://eutils.ncbi.nlm.nih.gov/entrez/eutils/';
 
   private $database = self::DEFAULT_DATABASE;
 
