diff --git a/sites/all/modules/contrib/apachesolr/apachesolr.module b/sites/all/modules/contrib/apachesolr/apachesolr.module
index f2f7cd69b..48ce52744 100644
--- a/sites/all/modules/contrib/apachesolr/apachesolr.module
+++ b/sites/all/modules/contrib/apachesolr/apachesolr.module
@@ -1512,14 +1512,9 @@ function apachesolr_do_query(DrupalSolrQueryInterface $current_query) {
   $query = apachesolr_current_query($env_id, $current_query);
 
   // Verify if this query was already executed in the same page load
-  if ($response = apachesolr_static_response_cache($searcher)) {
+  if ($response = apachesolr_static_response_cache($searcher) &! batch_get()) {
     // Return cached query object
-    if (batch_get()){
-      watchdog('ben_debug', 'IS_BATCH');
-    }else{
-      watchdog('ben_debug', 'NOT_BATCH');
-    }
-    // return array($query, $response);
+    return array($query, $response);
   }
   $query->addParam('start', $query->page * $query->getParam('rows'));
 
