diff --git a/sites/all/modules/contrib/biblio/modules/CiteProc/CSL.inc b/sites/all/modules/contrib/biblio/modules/CiteProc/CSL.inc
index 8651977..4c80caf 100644
--- a/sites/all/modules/contrib/biblio/modules/CiteProc/CSL.inc
+++ b/sites/all/modules/contrib/biblio/modules/CiteProc/CSL.inc
@@ -791,7 +791,7 @@ class csl_names extends csl_format {
     }
   }
 
-  function render($data, $mode) {
+  function render($data, $mode = NULL) {
     $matches = 0;
     $variable_parts = array();
     if (!isset($this->delimiter)) {
@@ -930,7 +930,7 @@ class csl_date extends csl_format {
 
   }
 
-  function render($data, $mode) {
+  function render($data, $mode = NULL) {
     $date_parts = array();
     $date = '';
     $text = '';
@@ -952,7 +952,7 @@ class csl_date extends csl_format {
 
 class csl_date_part extends csl_format {
 
-  function render($date, $mode) {
+  function render($date, $mode = FALSE) {
     $text = '';
 
     switch ($this->name) {
@@ -1000,7 +1000,7 @@ class csl_date_part extends csl_format {
 
 class csl_number extends csl_format {
 
-  function render($data, $mode) {
+  function render($data, $mode = NULL) {
     $var = $this->variable;
 
     if (!$var || empty($data->$var)) return;
@@ -1206,7 +1206,7 @@ class csl_macros extends csl_collection{
 
 class csl_group extends csl_format{
 
-  function render($data, $mode) {
+  function render($data, $mode = NULL) {
     $text = '';
     $text_parts = array();
 
@@ -1253,7 +1253,7 @@ class csl_layout extends csl_format {
     parent::init_formatting();
   }
 
-  function render($data, $mode) {
+  function render($data, $mode = NULL) {
     $text = '';
     $parts = array();
    // $delimiter = $this->delimiter;
