--- biblio.views.inc
+++ biblio.views.inc
@@ -86,12 +86,12 @@
       // links directly to node via vid
       'left_field' => 'vid',
       'field'      => 'vid',
-      'type'       => 'inner',
+      'type'       => 'left',
     ),
     'node_revision' => array(
       'left_field' => 'vid',
       'field'      => 'vid',
-      'type'       => 'inner',
+      'type'       => 'left',
     ),
     );
 
