diff --git a/includes/common.inc b/includes/common.inc
index 94363e6..144539b 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -5461,6 +5461,11 @@ function drupal_cron_cleanup() {
  *   - 'name': Name of file without the extension.
  */
 function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1) {
+  $cid = "drupal_system_listing|$mask|$directory|$key|$min_depth";
+  $cache = cache_get($cid);
+  if(!empty($cache->data)){
+    return $cache->data;
+  }
   $config = conf_path();
 
   $searchdir = array($directory);
@@ -5528,7 +5533,7 @@ function drupal_system_listing($mask, $directory, $key = 'name', $min_depth = 1)
     }
     $files = array_merge($files, $files_to_add);
   }
-
+  cache_set($cid, $files, 'cache', time() + 604800);
   return $files;
 }
 
@@ -7393,12 +7398,18 @@ function drupal_parse_info_file($filename) {
   $info = &drupal_static(__FUNCTION__, array());
 
   if (!isset($info[$filename])) {
-    if (!file_exists($filename)) {
-      $info[$filename] = array();
-    }
-    else {
-      $data = file_get_contents($filename);
-      $info[$filename] = drupal_parse_info_format($data);
+    $cache = cache_get(__FUNCTION__);
+    if(empty($cache->data) || empty($cache->data[$filename])){
+      if (!file_exists($filename)) {
+        $info[$filename] = array();
+      }
+      else {
+        $data = file_get_contents($filename);
+        $info[$filename] = drupal_parse_info_format($data);
+      }
+      cache_set(__FUNCTION__, $info, 'cache', time() + 604800);
+    } else {
+      $info = $cache->data;
     }
   }
   return $info[$filename];
