diff --git a/includes/common.inc b/includes/common.inc
index 43e2118..007967b 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -7401,7 +7401,7 @@ function entity_get_info($entity_type = NULL) {
   // Use the advanced drupal_static() pattern, since this is called very often.
   static $drupal_static_fast;
   if (!isset($drupal_static_fast)) {
-    $drupal_static_fast['entity_info'] = &drupal_static(__FUNCTION__);
+    $drupal_static_fast['entity_info'] = drupal_static(__FUNCTION__);
   }
   $entity_info = &$drupal_static_fast['entity_info'];
 
