--- sites/all/modules/contrib/migrate/migrate_ui/migrate_ui.pages.inc
+++ sites/all/modules/contrib/migrate/migrate_ui/migrate_ui.pages.inc
@@ -612,7 +612,8 @@
     foreach ($source->fields() as $machine_name => $description) {
       if (isset($source_key[$machine_name])) {
         // Identify primary key
-        $machine_name .= ' ' . t('(PK)');     
+        $machine_name .= ' ' . t('(PK)');
+        $classes = '';       
       }
       else {
         // Add class for mapped/unmapped. Used in summary.
