diff --git a/includes/database/mysql/schema.inc b/includes/database/mysql/schema.inc
index bf3ab86..62946aa 100644
--- a/includes/database/mysql/schema.inc
+++ b/includes/database/mysql/schema.inc
@@ -99,6 +99,8 @@ class DatabaseSchema_mysql extends DatabaseSchema {
 
     // Remove the last comma and space.
     $sql = substr($sql, 0, -3) . "\n) ";
+    
+    $sql .= 'ROW_FORMAT=DYNAMIC ';
 
     $sql .= 'ENGINE = ' . $table['mysql_engine'] . ' DEFAULT CHARACTER SET ' . $table['mysql_character_set'];
     // By default, MySQL uses the default collation for new tables, which is
