diff --git a/sites/all/modules/contrib/file_entity/file_entity.file_api.inc b/sites/all/modules/contrib/file_entity/file_entity.file_api.inc
index 339d91f..a6b7185 100644
--- a/sites/all/modules/contrib/file_entity/file_entity.file_api.inc
+++ b/sites/all/modules/contrib/file_entity/file_entity.file_api.inc
@@ -396,6 +396,11 @@ function file_displays_load($file_type, $view_mode, $key_by_formatter_name = FAL
   }
   $displays = ctools_export_load_object('file_display', 'names', $display_names);
 
+  $results = db_select('file_display', 'f')->fields('f')->condition('name', db_like($prefix).'%', 'LIKE')->execute();
+  foreach($results as $row){
+    $displays[$row->name] = $row;
+  }
+
   if ($key_by_formatter_name) {
     $prefix_length = strlen($prefix);
     $rekeyed_displays = array();
