--- taxonomy_breadcrumb.module
+++ taxonomy_breadcrumb.module
@@ -125,7 +125,7 @@
   $form['taxonomy_breadcrumb_path'] = array(
     '#type' => 'textfield',
     '#title' => t('Breadcrumb path (taxonomy_breadcrumb)'),
-    '#default_value' => _taxonomy_breadcrumb_get_vocabulary_path($form['vid']['#value']),
+    '#default_value' => isset($form['vid']['#value']) ? _taxonomy_breadcrumb_get_vocabulary_path($form['vid']['#value']) : '',
     '#maxlength' => 128,
     '#description' => t("Specify the path this vocabulary links to as a breadcrumb. If blank, the breadcrumb will not appear. Use a relative path and don't add a trailing slash. For example: node/42 or my/path/alias."),
     '#weight' => 0,
