--- drupal_selenium_web_test_case.php	Thu May 23 10:58:24 2013
+++ drupal_selenium_web_test_case.php	Thu May 23 11:04:42 2013
@@ -2322,7 +2322,7 @@
     $response = $this->execute("POST", "/session", $variables);
 
     // Parse out session id
-    preg_match("/\nLocation:.*\/(.*)\n/", $response['header'], $matches);
+    preg_match("/\nLocation:.*\/([^\n$]*)/", $response['header'], $matches);
     if (count($matches) > 0) {
       $this->session_id = trim($matches[1]);
     }
@@ -2365,7 +2365,7 @@
     // new line charachter.
     $response['header'] .= "\n";
     // Parse out session id
-    preg_match("/\n[Ll]ocation:.*\/(.*)\n/", $response['header'], $matches);
+    preg_match("/\n[Ll]ocation:.*\/([^\n$]*)/", $response['header'], $matches);
     if (count($matches) > 0) {
       $this->session_id = trim($matches[1]);
     }
@@ -2402,7 +2402,7 @@
     $response = $this->execute("POST", "/session", $variables);
 
     // Parse out session id
-    preg_match("/\nLocation:.*\/(.*)\n/", $response['header'], $matches);
+    preg_match("/\nLocation:.*\/([^\n$]*)/", $response['header'], $matches);
     if (count($matches) > 0) {
       $this->session_id = trim($matches[1]);
     }
@@ -2439,7 +2439,7 @@
     $response = $this->execute("POST", "/session", $variables);
 
     // Parse out session id
-    preg_match("/\nLocation:.*\/(.*)\n/", $response['header'], $matches);
+    preg_match("/\nLocation:.*\/([^\n$]*)/", $response['header'], $matches);
     if (count($matches) > 0) {
       $this->session_id = trim($matches[1]);
     }
