diff --git a/sites/all/modules/custom/darwincore/darwincore.features.field.inc b/sites/all/modules/custom/darwincore/darwincore.features.field.inc
index a8e8a75..7574848 100644
--- a/sites/all/modules/custom/darwincore/darwincore.features.field.inc
+++ b/sites/all/modules/custom/darwincore/darwincore.features.field.inc
@@ -1475,7 +1475,7 @@ function darwincore_field_default_fields(){
       'bundle' => 'specimen_observation',
       'default_value' => NULL,
       'deleted' => '0',
-      'description' => 'The catalogue number/identifier for this specimen',
+      'description' => 'The catalogue number/identifier for this specimen.  If left blank, a UUID will be generated for this specimen.',
       'display' => array(
         'default' => array(
           'label' => 'above',
@@ -1500,7 +1500,7 @@ function darwincore_field_default_fields(){
       'entity_type' => 'node',
       'field_name' => 'field_catalogue_number',
       'label' => 'Catalogue number',
-      'required' => 1,
+      'required' => 0,
       'settings' => array(
         'text_processing' => '0',
         'user_register_form' => FALSE
@@ -1549,7 +1549,7 @@ function darwincore_field_default_fields(){
       'bundle' => 'specimen_observation',
       'default_value' => NULL,
       'deleted' => '0',
-      'description' => 'The code (or acronym) identifying the collection within the institution in which the organism record is catalogued.',
+      'description' => 'The code (or acronym) identifying the collection within the institution in which the organism record is catalogued. If left blank, this field will be populated with "N/A".',
       'display' => array(
         'default' => array(
           'label' => 'above',
@@ -1574,7 +1574,7 @@ function darwincore_field_default_fields(){
       'entity_type' => 'node',
       'field_name' => 'field_collection_code',
       'label' => 'Collection code',
-      'required' => 1,
+      'required' => 0,
       'settings' => array(
         'text_processing' => '0',
         'user_register_form' => FALSE
diff --git a/sites/all/modules/custom/darwincore/darwincore.module b/sites/all/modules/custom/darwincore/darwincore.module
index 868dad6..dd53bf6 100644
--- a/sites/all/modules/custom/darwincore/darwincore.module
+++ b/sites/all/modules/custom/darwincore/darwincore.module
@@ -150,6 +150,13 @@ function darwincore_specimen_observation_node_form_submit($form, &$form_state){
 function darwincore_specimen_observation_node_form_validate($form, &$form_state){
   // Because we hold the data in separate fields (yay for the efficient fields
   // modules!), we'll start with the catalogue number and work up.
+  if(empty($form_state['values']['field_catalogue_number'][LANGUAGE_NONE][0]['value'])){
+    $form_state['values']['field_catalogue_number'][LANGUAGE_NONE][0]['value'] = uuid_generate();
+    $form_state['node']->uuid = $form_state['values']['field_catalogue_number'][LANGUAGE_NONE][0]['value'];
+  }
+  if(empty($form_state['values']['field_collection_code'][LANGUAGE_NONE][0]['value'])){
+    $form_state['values']['field_collection_code'][LANGUAGE_NONE][0]['value'] = 'N/A';
+  }
   $count = db_query('SELECT COUNT(*) FROM {field_data_field_catalogue_number} WHERE field_catalogue_number_value = :value AND entity_id != :nid', array(
     ':value' => $form_state['values']['field_catalogue_number'][LANGUAGE_NONE][0]['value'],
     ':nid' => isset($form_state['values']['nid']) ? $form_state['values']['nid'] : 0
