diff --git a/sites/all/modules/contrib/ckeditor/plugins/counter/plugin.js b/sites/all/modules/contrib/ckeditor/plugins/counter/plugin.js
index 28ce728..2120b42 100644
--- a/sites/all/modules/contrib/ckeditor/plugins/counter/plugin.js
+++ b/sites/all/modules/contrib/ckeditor/plugins/counter/plugin.js
@@ -33,6 +33,9 @@
 
       function count( ev )
       {
+        if (!ev.editor.document) {
+          return;
+        }
         var space = getSpaceElement();
         var text = ev.editor.getData();
         // decode HTML entities; it also removes HTML tags, but works only if jQuery is available
