diff --git a/sites/all/modules/contrib/menu_block/menu_block.module b/sites/all/modules/contrib/menu_block/menu_block.module
index 7010188..e56bcad 100644
--- a/sites/all/modules/contrib/menu_block/menu_block.module
+++ b/sites/all/modules/contrib/menu_block/menu_block.module
@@ -668,7 +668,7 @@ function menu_tree_sort_active_path(array &$tree) {
  * @param array $parent_item
  *   The menu item that should be used as the root of the tree.
  */
-function menu_tree_prune_tree(array &$tree, $level, array $parent_item = NULL) {
+function menu_tree_prune_tree(array &$tree, $level, $parent_item = NULL) {
   if (!empty($parent_item)) {
     // Prune the tree along the path to the menu item.
     for ($i = 1; $i <= MENU_MAX_DEPTH && $parent_item["p$i"] != '0'; $i++) {
