diff --git a/js/dependent.js b/js/dependent.js
index 07ff021..e9e2447 100644
--- a/js/dependent.js
+++ b/js/dependent.js
@@ -97,7 +97,7 @@
           else {
             switch ($(trigger).attr('type')) {
               case 'checkbox':
-                var val = $(trigger).attr('checked') || 0;
+                var val = $(trigger).attr('checked') ? true : false;
 
                 if (val) {
                   $(trigger).siblings('label').removeClass('hidden-options').addClass('expanded-options');
