Index: /lib/OAuth.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- /lib/OAuth.php	(date 1328724051000)
+++ /lib/OAuth.php	(revision )
@@ -1,10 +1,13 @@
 <?php
 // vim: foldmethod=marker
 
-/* Generic exception class
+if (!class_exists('OAuthException')) {
+  /*
+   * Generic exception class
- */
-class OAuthException extends Exception {
-  // pass
+   */
+  class OAuthException extends Exception {
+    // pass
+  }
 }
 
 class OAuthConsumer {
