diff --git a/sites/all/libraries/PHPExcel/PHPExcel/Shared/PCLZip/pclzip.lib.php b/sites/all/libraries/PHPExcel/PHPExcel/Shared/PCLZip/pclzip.lib.php
index 5b81e6790..1fa853d94 100644
--- a/sites/all/libraries/PHPExcel/PHPExcel/Shared/PCLZip/pclzip.lib.php
+++ b/sites/all/libraries/PHPExcel/PHPExcel/Shared/PCLZip/pclzip.lib.php
@@ -212,7 +212,7 @@
   //   Note that no real action is taken, if the archive does not exist it is not
   //   created. Use create() for that.
   // --------------------------------------------------------------------------------
-  function PclZip($p_zipname)
+  function __construct($p_zipname)
   {
 
     // ----- Tests the zlib
@@ -1836,6 +1836,7 @@
     $v_memory_limit = ini_get('memory_limit');
     $v_memory_limit = trim($v_memory_limit);
     $last = strtolower(substr($v_memory_limit, -1));
+    $v_memory_limit = intval($v_memory_limit);
  
     if($last == 'g')
         //$v_memory_limit = $v_memory_limit*1024*1024*1024;
