diff --git a/includes/module.inc b/includes/module.inc
index 3a019f2..5271c0f 100644
--- a/includes/module.inc
+++ b/includes/module.inc
@@ -421,10 +421,10 @@ function module_enable($module_list, $enable_dependencies = TRUE) {
       module_list(TRUE);
       module_implements('', FALSE, TRUE);
       _system_update_bootstrap_status();
-      // Update the registry to include it.
-      registry_update();
       // Refresh the schema to include it.
       drupal_get_schema(NULL, TRUE);
+      // Update the registry to include it.
+      registry_update();
       // Clear entity cache.
       entity_info_cache_clear();
 
